pytest/common_lib/shared/__init__.py

```diff
@@ -98,56 +105,68 @@ def start_neard_cluster_with_cleanup(
+) -> List[Candidate]:
+    """
+    Generate MPC configs for each participant.
+    Without loss of generality, we will make all MPC participant's near account a subaccount of the main (contract) node.
+    This will make things easier. Otherwise:
+    FIXME: the canonical way is to create completely new accounts via registrar account.
+      (1) How to get it via py api?
+      (2) observer nodes that corresponds to the mpc participant hasn't been started yet,
+        so we can not make any requests from them yet.
+    """
+    signers = ','.join(f'signer_{i}.test0' for i in range(num_mpc_nodes))
+    responders = [f'responder_{i}.test0' for i in range(num_mpc_nodes)]
     cmd = (
         MPC_BINARY_PATH, 'generate-test-configs',
         '--output-dir', dot_near,
-        '--participants', participants,
-        '--threshold', str(num_mpc_nodes)
+        '--participants', signers,
+        '--responders', ','.join(responders),
+        '--threshold', str(num_mpc_nodes),
+        '--desired-responder-keys-per-participant', str(num_respond_aks),
     )
     if presignatures_to_buffer:
         cmd = cmd + (
             '--desired-presignatures-to-buffer', str(presignatures_to_buffer),
         )
     subprocess.run(cmd)
 
+    candidates = []
+    with open(pathlib.Path(dot_near / 'participants.json')) as file:
+        participants_config = yaml.load(file, Loader=SafeLoaderIgnoreUnknown)
+    for idx, (participant, responder_account_id) in enumerate(zip(participants_config['participants'], responders)):
+        near_account = participant['near_account_id']
+        p2p_public_key = participant['p2p_public_key']
+        my_addr = participant['address']
+        my_port = participant['port']
+
+        secrets_file_path = os.path.join(dot_near, str(idx), 'secrets.json')
+        with open(secrets_file_path) as file:
+            participant_secrets = json.load(file)
+        signer_key = deserialize_key(near_account, participant_secrets['near_signer_key'])
+        responder_keys = []
+        for key in participant_secrets['near_responder_keys']:
+            responder_keys.append(deserialize_key(responder_account_id, key))
 
-def create_and_dump_responder_config(
-        num_respond_aks: int,
-        mpc_node: MpcNode,
-        # We can not get `last_block_hash` from non-validator node at this moment, because it hasn't been started yet.
-        # So we send transactions via validator node.
-        validator_account: NearAccount,
-):
-    if num_respond_aks == 0:
-        return
-    last_block_hash = validator_account.last_block_hash()
-    account_id = f"respond.{mpc_node.account_id()}"
-    access_keys = [
-        Key.from_seed_testonly(account_id, seed=f"{s}")
-        for s in range(0, num_respond_aks)
-    ]
-    tx = sign_create_account_with_multiple_access_keys_tx(
-        mpc_node.signer_key(),
-        account_id,
-        access_keys,
-        1,
-        last_block_hash
-    )
```

Comment:
I thought we would still need this logic, just that the `access_keys` are read from the nodes debug endpoint.